File: frontend/src/types/graph.ts
export interface GraphNode {
  id: string;
  type: "journal" | "metric_type" | "exercise_type" | "goal";
  label: string;
  tags: string[];
  date?: string;
  status?: string;
  progress?: number;
}

export interface GraphEdge {
  source: string;
  target: string;
  type: "mentions" | "tracks" | "shared_tag";
  tag?: string;
}

export interface GraphResponse {
  nodes: GraphNode[];
  edges: GraphEdge[];
}

File: backend/app/schemas/graph.py
"""Schemas for the graph visualization API."""

from __future__ import annotations

from pydantic import BaseModel


class GraphNode(BaseModel):
    id: str
    type: str
    label: str
    tags: list[str] = []
    # Optional fields depending on type
    date: str | None = None
    status: str | None = None
    progress: int | None = None

    model_config = {"from_attributes": True}


class GraphEdge(BaseModel):
    source: str
    target: str
    type: str
    tag: str | None = None

    model_config = {"from_attributes": True}


class GraphResponse(BaseModel):
    nodes: list[GraphNode]
    edges: list[GraphEdge]
